tools/kprint/main.cpp

```diff
@@ -1,33 +1,36 @@
 #include "kllvm/ast/AST.h"
 #include "kllvm/printer/printer.h"
+#include "llvm/Support/CommandLine.h"
+#include "llvm/Support/WithColor.h"
 
 #include <iostream>
 #include <string>
 #include <unistd.h>
 
 using namespace kllvm;
+using namespace llvm;
+
+cl::opt<std::string> DefinitionFilename(
+    cl::Positional, cl::desc("<definition.kore>"), cl::Required);
+
+cl::opt<std::string>
+    PatternFilename(cl::Positional, cl::desc("<pattern.kore>"), cl::Required);
+
+cl::opt<std::string> FilterSubst(cl::Positional, cl::desc("[true|false]"));
+
+cl::opt<std::string> ArgColor(cl::Positional, cl::desc("[true|false|auto]"));
```

Comment:
Again rather than a string, let's make this a dedicated enum with three values.

Reply:
I don't think we can use `cl::Positional` with enums because `clEnumVal` and `clEnumValN` creates a list of "GeneralOptions" like:
```
USAGE: kprint [options] <definition.kore> <pattern.kore> Colorize output [true|false]

OPTIONS:

General options:

  Colorize output
      --true     - always colorize
      --false    - never colorize
      --auto     - colorize if stdout is a tty
```

This improves the documentation for this additional parameter. The downside is that the API is modified to use `--true` instead of `true`.

Reply:
Sure, looking at the documentation that does indeed seem to be the case. I see two options:
* Specialize `llvm::cl::parser` for your enum
* Validate by hand that the option is in `{"true", "false", "auto"}`.

Either of these is probably fine; weak preference for the first to get LLVM's parse error handling, but if the code is problematic then don't worry.

Reply:
https://llvm.org/docs/CommandLine.html#writing-a-custom-parser ah, you don't get parse errors for free. Just validate the string in that case